In [2]:
import numpy as np
import pandas as pd
import functions

In [3]:
local_path = "data/Partido_06032025_QuimsavsIACC_visitante.csv" #IACC
visit_path = "data/Partido_06032025_QuimsavsIACC_local.csv" #QUIMSA
functions.four_factors(local_path, visit_path)

(0.46774193548387094,
 0.38235294117647056,
 0.2115812917594655,
 0.1907587961000424,
 0.25,
 0.75,
 0.16129032258064516,
 0.16176470588235295)

In [4]:
functions.show_four_factor_tabla(functions.four_factors(local_path, visit_path))

           shooting  turnover  rebounding  Free throws
offensive  0.467742  0.211581        0.25     0.161290
defensive  0.382353  0.190759        0.75     0.161765


# 1. Shooting (Eficiencia de campo - eFG%):
**Ofensivo (0.467742):** Este valor significa que el equipo tiene una eficiencia de campo ofensiva del 46.77%. Es un valor moderado, lo que sugiere que hay margen para mejorar la efectividad en los tiros.

**Defensivo (0.382353):** El equipo está permitiendo a sus oponentes una eficiencia de campo defensiva de 38.24%. Esto es un excelente indicador, ya que significa que están defendiendo bien y limitando los tiros efectivos del rival.

# 2. Turnover (Pérdidas de balón - TOV%):
**Ofensivo (0.211581):** Un valor de 21.16% de pérdidas de balón en ofensiva es bastante alto, lo que sugiere que el equipo necesita mejorar el control del balón para evitar desperdiciar posesiones.

**Defensivo (0.190759):** Un 19.08% de pérdidas de balón forzadas en defensa es un valor sólido, indicando que el equipo está presionando bien al rival y logrando recuperar el balón con frecuencia.

# 3. Rebounding (Rebotes - REB%):
**Ofensivo (0.25):** El equipo está obteniendo el 25% de los rebotes ofensivos disponibles. Es un valor aceptable, aunque podría ser mejorado para generar más segundas oportunidades de anotación.

**Defensivo (0.75):** Un 75% de los rebotes defensivos es un buen porcentaje, aunque menor que el del análisis previo. Esto indica que el equipo sigue controlando bien el cristal defensivo, pero con margen para mejorar.

# 4. Free Throws (Tiros libres - FT%):
**Ofensivo (0.161290):** El equipo está obteniendo 16.13% de sus puntos a través de tiros libres. Es un valor bajo, lo que sugiere que podrían buscar más oportunidades para generar faltas y convertir desde la línea.

**Defensivo (0.161765):** El equipo está permitiendo al oponente un 16.18% de sus puntos a través de tiros libres. Este es un buen indicador, ya que significa que están cometiendo pocas faltas y limitando los puntos del rival desde la línea.

# Resumen:
## Ofensivo:
- El equipo tiene una eficiencia de tiro moderada.
- Necesita mejorar el control de pérdidas de balón, ya que el porcentaje de pérdidas es alto.
- Captura un número aceptable de rebotes ofensivos.
- Genera pocos puntos desde la línea de tiros libres.

## Defensivo:
- Tiene una excelente defensa en tiros, limitando la eficiencia de campo del oponente.
- Fuerza pérdidas de balón a un ritmo sólido.
- Mantiene un buen control de los rebotes defensivos, aunque con margen de mejora.
- Comete pocas faltas, evitando conceder demasiados puntos desde la línea de tiros libres.

En general, el equipo muestra una defensa muy fuerte, especialmente en la eficiencia de tiro del rival y en la capacidad de generar pérdidas de balón. En ofensiva, deben mejorar el control del balón y generar más oportunidades de anotación mediante rebotes ofensivos y tiros libres.

